src/v/redpanda/debug_bundle.cc

```diff
@@ -0,0 +1,292 @@
+
+#include "config/configuration.h"
+#include "ssx/future-util.h"
+#include "utils/gate_guard.h"
+#include "utils/string_switch.h"
+#include "vlog.h"
+
+#include <seastar/core/loop.hh>
+#include <seastar/core/reactor.hh>
+#include <seastar/http/exception.hh>
+#include <seastar/util/log.hh>
+#include <seastar/util/process.hh>
+
+#include <fmt/format.h>
+
+#include <filesystem>
+#include <optional>
+
+static ss::logger bundle_log{"debug_bundle"};
+
+using consumption_result_type
+  = debug_bundle::rpk_consumer::consumption_result_type;
+using stop_consuming_type = debug_bundle::rpk_consumer::stop_consuming_type;
+using tmp_buf = stop_consuming_type::tmp_buf;
+
+ss::future<consumption_result_type>
+debug_bundle::rpk_consumer::operator()(tmp_buf buf) {
+    std::string str_buf{buf.begin(), buf.end()}, line;
+    vlog(bundle_log.trace, "Consumer read {}", str_buf);
+    std::stringstream ss(str_buf);
+    while (!ss.eof()) {
+        std::getline(ss, line);
+        if (line.starts_with("Debug bundle saved to")) {
+            vlog(bundle_log.trace, "Stop condition reached on line {}", line);
+            status = debug_bundle_status::not_running;
+            return make_ready_future<consumption_result_type>(
+              stop_consuming_type({}));
+        }
+    }
+    return make_ready_future<consumption_result_type>(ss::continue_consuming{});
+}
+
+constexpr std::string_view to_string_view(debug_bundle_status bundle_status) {
+    switch (bundle_status) {
+    case debug_bundle_status::not_running:
+        return "not-running";
+    case debug_bundle_status::running:
+        return "running";
+    }
+    return "unknown";
+}
+
+template<typename E>
+std::enable_if_t<std::is_enum_v<E>, std::optional<E>>
+  from_string_view(std::string_view);
+
+template<>
+constexpr std::optional<debug_bundle_status>
+from_string_view<debug_bundle_status>(std::string_view sv) {
+    return string_switch<std::optional<debug_bundle_status>>(sv)
+      .match(
+        to_string_view(debug_bundle_status::not_running),
+        debug_bundle_status::not_running)
+      .match(
+        to_string_view(debug_bundle_status::running),
+        debug_bundle_status::running)
+      .default_match(std::nullopt);
+}
+
+debug_bundle::debug_bundle(
+  const std::filesystem::path& write_dir, const std::filesystem::path& rpk_path)
+  : _status{debug_bundle_status::not_running}
+  , _write_dir{write_dir}
+  , _in_progress_filename{"debug-bundle.zip"}
+  , _rpk_cmd{rpk_path}
+  , _transfer_handle{
+      detail::make_bundle_filename(get_write_dir(), _in_progress_filename),
+      nullptr,
+      false} {}
+
+ss::future<> debug_bundle::stop() {
+    using namespace std::chrono_literals;
+    vlog(bundle_log.info, "Stopping debug bundle ...");
+    co_await _rpk_gate.close();
+}
+
+ss::future<> debug_bundle::start_creating_bundle(
+  const request_auth_result& auth_state, const debug_bundle_params params) {
+    if (_status == debug_bundle_status::running) {
```

Comment:
this doesn't look safe. what happens if this check is made and then another request arrives to create the bundle before the status is updated?

Reply:
good catch, I'm re-writting this for clarity:

Assume status is "not_running".
1. Request A to start debug bundle comes in, passes the check but RP has not yet updated the status
2. Request B to start debug bundle comes in, and also passes the check since the status is not updated yet.